ethcontract-common/src/bytecode.rs

```diff
@@ -17,20 +17,31 @@ use web3::types::{Address, Bytes};
 pub struct Bytecode(String);
 
 impl Bytecode {
-    /// Reads hex bytecode representation from a string slice.
+    /// Reads hex bytecode representation from a string or map slice.
     pub fn from_hex_str(s: &str) -> Result<Self, BytecodeError> {
         if s.is_empty() {
             // special case where we have an empty string byte code.
             return Ok(Bytecode::default());
         }
 
+        let s = if s.starts_with('{') {
+            #[derive(Deserialize)]
```

Comment:
This needs some explanation in the code like what you put in the PR description. Should explain why we have this if, what it does, and if possible link to the foundry documentation of their output (not sure this exists).